addons/hr_expense/models/account_payment.py

```diff
@@ -33,7 +33,8 @@ def _synchronize_from_moves(self, changed_fields):
 
     def _synchronize_to_moves(self, changed_fields):
         # EXTENDS account
-        if self.expense_sheet_id:
+        trigger_fields = self._get_trigger_fields_to_synchronize()
+        if self.expense_sheet_id and any(field_name in trigger_fields for field_name in changed_fields):
```

Comment:
This allows to edit fields on the payment that should not be editable
(the payment method & the memo)

Reply:
No reason to whitelist imo. But do ban those two fields

Reply:
> (the payment method & the memo)

It seems payment method is already not editable after creating the payment (at least for this specific use case)
I've blocked the edit of memo and expense sheet

Reply:
It is, with your fix if you set the payment to draft...